integration_test/common/k8s_specific.go

```diff
@@ -0,0 +1,146 @@
+func IsPodStateWait(podState string) bool {
+	for _, state := range PodWaitStates {
+		if state == podState {
+			return true
+		}
+	}
+
+	return false
+}
+
+// IsPodStateGood checks if supplied pod state is good or not
+// by matching state of supplied pod state with pre-identified Good states list (PodGoodStates)
+func IsPodStateGood(podState string) bool {
+	for _, state := range PodGoodStates {
+		if state == podState {
+			return true
+		}
+	}
+
+	return false
+}
+
+// GetClientConfig first tries to get a config object which uses the service account kubernetes gives to pods,
+// if it is called from a process running in a kubernetes environment.
+// Otherwise, it tries to build config from a default kubeconfig filepath if it fails, it fallback to the default config.
+// Once it get the config, it returns the same.
+func GetClientConfig() (*rest.Config, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		if debug {
+			fmt.Printf("Unable to create config. Error: %+v", err)
+		}
+		err1 := err
+		kubeconfig := filepath.Join(os.Getenv("HOME"), ".kube", "config")
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+		if err != nil {
+			err = fmt.Errorf("InClusterConfig as well as BuildConfigFromFlags Failed. Error in InClusterConfig: %+v\nError in BuildConfigFromFlags: %+v", err1, err)
+			return nil, err
+		}
+	}
+
+	return config, nil
+}
+
+// GetClientset first tries to get a config object which uses the service account kubernetes gives to pods,
+// if it is called from a process running in a kubernetes environment.
+// Otherwise, it tries to build config from a default kubeconfig filepath if it fails, it fallback to the default config.
+// Once it get the config, it creates a new Clientset for the given config and returns the clientset.
+func GetClientset() (*kubernetes.Clientset, error) {
+	config, err := GetClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		err = fmt.Errorf("Failed creating clientset. Error: %+v", err)
+		return nil, err
+	}
+
+	return clientset, nil
+}
+
+// GetRESTClient first tries to get a config object which uses the service account kubernetes gives to pods,
+// if it is called from a process running in a kubernetes environment.
+// Otherwise, it tries to build config from a default kubeconfig filepath if it fails, it fallback to the default config.
+// Once it get the config, it
+func GetRESTClient() (*rest.RESTClient, error) {
+	config, err := GetClientConfig()
+	if err != nil {
+		return &rest.RESTClient{}, err
+	}
+
+	return rest.RESTClientFor(config)
+}
+
+// GetCoreV1API returns a k8s.io/client-go/kubernetes/typed/core/v1.CoreV1Interface object
+// and nil as error in case of no error occures when it tries to make a clientset
+// otherwise it retures nil for k8s.io/client-go/kubernetes/typed/core/v1.CoreV1Interface
```

Comment:
Typo mistake !
retures should be returns !

Reply:
Addressed.